projects/storefrontlib/src/cms-structure/outlet/outlet.directive.ts

```diff
@@ -58,17 +60,31 @@ export class OutletDirective implements OnInit, OnDestroy {
   ) {}
 
   ngOnInit(): void {
+    this.initializeOutlet();
+  }
+
+  private initializeOutlet(): void {
+    this.vcr.clear();
+    this.subscription.unsubscribe();
+    this.subscription = new Subscription();
+
     if (this.cxOutletDefer) {
       this.deferLoading();
     } else {
       this.render();
     }
   }
 
-  private deferLoading() {
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.cxOutlet && !changes.cxOutlet.firstChange) {
```

Comment:
Can we simplify it? 
- remove the condition for `.firstChange`
- remove the hook `ngOnInit`?

Reply:
Good idea